core/trino-main/src/test/java/io/trino/server/testing/TestTestingTrinoServer.java

```diff
@@ -63,4 +64,26 @@ void testSetCatalogManagementToStatic()
                     .isInstanceOf(StaticCatalogManager.class);
         }
     }
+
+    @Test
+    void testDefaultCatalogStore()
+            throws IOException
+    {
+        try (TestingTrinoServer server = TestingTrinoServer.builder().build()) {
+            assertThat(server.getInstance(Key.get(CatalogStore.class)))
+                    .isInstanceOf(InMemoryCatalogStore.class);
+        }
+    }
+
+    @Test
+    void testExplicitCatalogStore()
+            throws IOException
+    {
+        try (TestingTrinoServer server = TestingTrinoServer.builder()
+                .addProperty("catalog.store", "file")
+                .build()) {
+            assertThat(server.getInstance(Key.get(CatalogStore.class)))
+                    .isInstanceOf(FileCatalogStore.class);
+        }
+    }
```

Comment:
You should test the Dynamic custom store. 
In invoking a New plugin based on catalogstore catalogstorefactory and plugin interface that contains the getcatalogstorefactoriesn method.

My last test in your code configuration let me confuse. I m not sure that it works well on a custom plugin for catalog store.

Reply:
Thanks